src/hid/hid-device.cpp

```diff
@@ -195,7 +215,27 @@ namespace librealsense
         {
             REALSENSE_HID_REPORT report;
 #ifdef __APPLE__
-            hid_read(_hidapi_device, reinterpret_cast<unsigned char*>(&report), sizeof(REALSENSE_HID_REPORT));
+            unsigned char tmp_buffer[100] = { 0 };
+            hid_read( _hidapi_device, tmp_buffer, _realsense_hid_report_actual_size );
+            if( _realsense_hid_report_actual_size != sizeof( REALSENSE_HID_REPORT ) )
+            {
+                // for FX version<5.16 the actual struct is 32 bit, so we can not use memcpy for the whole struct
+                // for FX version<5.16 x,y,x are all short with the x variable located at offset 10, y at offset 12, and
+                // z at offset 14 within the structure.
+                memcpy( &report, tmp_buffer, 10 );
```

Comment:
We have 2 structs right?
Where x,y,z are 16 and when there are 32?
If so, can we use `std::offsetof` here?
Like we do in the metadata parsing..

Reply:
Example: https://github.com/IntelRealSense/librealsense/blob/43133ff7c9628899394fd6a6686c7550bb2346f0/src/ds/d500/d500-color.cpp#L161